pkg/conversation/conversation.go

```diff
@@ -0,0 +1,356 @@
+
+import (
+	piiscrubber "github.com/aavaz-ai/pii-scrubber"
+	contribConverse "github.com/dapr/components-contrib/conversation"
+	"github.com/dapr/dapr/pkg/messages"
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/kit/ptr"
+)
+
+// ConvertUsageToProto converts components-contrib UsageInfo to Dapr protobuf ConversationUsage
+func ConvertUsageToProto(resp *contribConverse.ConversationResponse) *runtimev1pb.ConversationUsage {
+	if resp == nil || resp.Usage == nil {
+		return nil
+	}
+
+	// Safely convert uint64 to uint32 for token counts
+	// Token counts realistically never exceed uint32 range (~4.3 billion)
+	return &runtimev1pb.ConversationUsage{
+		PromptTokens:     ptr.Of(uint32(resp.Usage.PromptTokens)),
+		CompletionTokens: ptr.Of(uint32(resp.Usage.CompletionTokens)),
+		TotalTokens:      ptr.Of(uint32(resp.Usage.TotalTokens)),
+	}
+}
+
+// NeedsInputScrubber checks if PII scrubbing is needed for any input
+func NeedsInputScrubber(req *runtimev1pb.ConversationRequest) bool {
+	for _, input := range req.GetInputs() {
+		if input.GetScrubPII() {
+			return true
+		}
+	}
+	return false
+}
+
+// NeedsOutputScrubber checks if PII scrubbing is needed for outputs
+func NeedsOutputScrubber(req *runtimev1pb.ConversationRequest) bool {
+	return req.GetScrubPII()
+}
+
+// ConvertProtoToolsToComponentsContrib converts protobuf tools to components-contrib format
+func ConvertProtoToolsToComponentsContrib(protoTools []*runtimev1pb.Tool) []contribConverse.Tool {
+	if len(protoTools) == 0 {
+		return nil
+	}
+
+	tools := make([]contribConverse.Tool, len(protoTools))
+	for i, protoTool := range protoTools {
+		tool := contribConverse.Tool{
+			ToolType: protoTool.GetType(),
+			Function: contribConverse.ToolFunction{
+				Name:        protoTool.GetName(),
+				Description: protoTool.GetDescription(),
+				Parameters:  protoTool.GetParameters(), // Keep as string for now
+			},
+		}
+		tools[i] = tool
+	}
+
+	return tools
+}
+
+// ConvertComponentsContribToolCallsToProto converts components-contrib tool calls to protobuf format
+func ConvertComponentsContribToolCallsToProto(componentToolCalls []contribConverse.ToolCall) []*runtimev1pb.ToolCall {
+	if len(componentToolCalls) == 0 {
+		return nil
+	}
+
+	protoToolCalls := make([]*runtimev1pb.ToolCall, len(componentToolCalls))
+	for i, componentToolCall := range componentToolCalls {
+		protoToolCall := &runtimev1pb.ToolCall{
+			Id:        componentToolCall.ID,
+			Type:      componentToolCall.CallType,
+			Name:      componentToolCall.Function.Name,
+			Arguments: componentToolCall.Function.Arguments,
+		}
+		protoToolCalls[i] = protoToolCall
+	}
+
+	return protoToolCalls
+}
+
+// extractToolCallIDFromParts extracts tool call ID from tool result parts
+func extractToolCallIDFromParts(parts []*runtimev1pb.ContentPart) string {
+	for _, part := range parts {
+		if toolResult := part.GetToolResult(); toolResult != nil {
+			return toolResult.GetToolCallId()
+		}
+	}
+	return ""
+}
```

Comment:
Same here- why no error, and why do we only return _one_ tool id?

Reply:
no idea